src/SFA.DAS.Tools.Servicebus.Support.Infrastructure/Services/CosmosDb/CosmosMessageDbContext.cs

```diff
@@ -222,5 +223,24 @@ private string AddPaging(string sqlQuery, SearchProperties searchProperties)
         }
 
         private static string AddTypeClause(string sqlQuery) => sqlQuery + " AND c.type='message'";
+        
+        public async Task<IEnumerable<UserMessageCount>> GetMessageCountPerUser()
+        {            
+            var sqlQuery = $"select value messages  from (select c.Queue, c.userId,  count(1) as MessageCount from c where c.type = 'message' group by c.Queue, c.userId ) as messages";
+
+            var queryFeedIterator = await QuerySetup<UserMessageCount>(sqlQuery);
+
+            var messages = new List<UserMessageCount>();
+
+            while (queryFeedIterator.HasMoreResults)
+            {
+                var currentResults = await queryFeedIterator.ReadNextAsync();
+
+                var newMessages = currentResults.ToList();
+                messages.AddRange(newMessages);
```

Comment:
Combine these two lines and remove the need for newMessages var
```suggestion
                messages.AddRange(currentResults.ToList());
```